agrirouter-sdk-dotnet-standard-impl/Service/Messaging/Vcu/OnboardVcuService.cs

```diff
@@ -87,5 +90,39 @@ public EncodedMessage Encode(OnboardVcuParameters onboardVcuParameters)
 
             return encodedMessage;
         }
+
+        public OnboardingResponse Decode(Any messageResponse)
+        {
+            try
+            {
+                return OnboardingResponse.Parser.ParseFrom(messageResponse.Value);
+            }
+            catch (Exception e)
+            {
+                throw new CouldNotDecodeMessageException(
+                    "Could not decode onboard response for virtual communication unit.", e);
+            }
+        }
+
+
+        /// <summary>
+        /// Please see <seealso cref="IOnboardVcuService.EnhanceVirtualCommunicationToFullyUsableOnboardResponse" /> for documentation.
+        /// </summary>
+        /// <param name="parent"></param>
+        /// <param name="virtualCommunicationUnit"></param>
+        /// <returns></returns>
+        public OnboardResponse EnhanceVirtualCommunicationToFullyUsableOnboardResponse(OnboardResponse parent,
+            OnboardingResponse.Types.EndpointRegistrationDetails virtualCommunicationUnit)
+        {
+            var capabilityAlternateId = virtualCommunicationUnit.CapabilityAlternateId;
+            var deviceAlternateId = virtualCommunicationUnit.DeviceAlternateId;
+            var sensorAlternateId = virtualCommunicationUnit.SensorAlternateId;
+
+            parent.CapabilityAlternateId = capabilityAlternateId;
+            parent.DeviceAlternateId = deviceAlternateId;
+            parent.SensorAlternateId = sensorAlternateId;
+
+            return parent;
```

Comment:
While using this I was expecting a new onboarding response for the VCU. This is changing the application onboarding response to be the VCU's isn't it? Why not create a new onboarding response?

Reply:
I guess I was assuming I would pass my applications onboarding response. Is that not what was intended?

Reply:
The VCU is tightly coupled to the endpoint it was created with, therefore the VCU is using the certificates of the endpoint (and all the other information required to connect). You are using the onboard response of the endpoint (which is an instance of the application / the telemetry platform) as a base and just change the identifying part for the VCU.

Reply:
I get what is needed now for the VCU onboarding response object. What I mean is if I pass this function my app's onboarding response to be converted into the VCU onboarding response, I no longer have a valid app onboarding response.

Reply:
It seems we have a major misunderstanding here. What do you mean by „no longer work“?
Situation before:
- you have an onboarding response for the telemetry platform
- you have the VCU details

Situation afterwards:
- you have an onboarding response for the telemetry platform
- you have an onboarding response for the VCU

i don’t understand why you assume that you would lose the former along the way?

Reply:
When the output type of a method is an object, I expect it to be a new object. This method is taking the onboarding response object, which in this case is the onboarding response for the telemetry platform, and it is modifying that object. In my case, I was taking the telemetry platform onboarding response object and passing it to this method before storing it. After the VCU onboarding I was then trying to store the telemetry platform onboard response and the vcu onboarding response. The problem is they are they same object now because all this method does is change the three parameters of the input onboarding response. If you don't store the telemetry platform onboarding response before calling this method, the device alt id, sensor alt id, and endpoint id are changed to the VCU values, i.e. I no longer have a valid onboarding response for my telemetry platform.

Reply:
@saschadoemer and @oliverrahner I sent you a VS debugging session video that might show it better.

Reply:
I got your point now, sorry... 
I was thinking along the lines of: you already have to have that onboarding response persisted, where's the problem?
I didn't pay attention that we actually change the passed object.

Reply:
It just needs a deep copy. A task for future me 😴

Reply:
@2Dews Please see the latest commit.